frontend/src/hooks/useJobs.tsx

```diff
@@ -26,5 +26,18 @@ export function useJobs() {
             .then(data => setJobs(prevState => [...prevState, data]))
     }
 
-    return {jobs, addJob, getAllJobs}
+    function deleteJobById(id: string | undefined) {
+        axios.delete("/api/jobs/" + id)
+            .then(response => response.data)
+            .then(data => {
+                setJobs(prevState => {
+                    const currentJobs = prevState.filter(job => job.id !== id)
```

Comment:
```suggestion
                    const currentJobs = prevState.filter(job => job.id !== data.id)
```

Like this you use the returned data, otherwise you could also drop the  `.then(response => response.data)`, as now you are not using it. But I suggest using it, so you are sure that backend and frontend stay in sync. (Please check if my suggested code is correct if you use it, I didn't test it :) )